internal/apiserver/server.go

```diff
@@ -341,20 +341,19 @@ func (as *apiServer) swaggerUIHandler(url string) func(res http.ResponseWriter,
 	}
 }
 
-func (as *apiServer) getAPIURL() string {
-	proto := "https"
-	if !apiConfigPrefix.GetBool(HTTPConfTLSEnabled) {
-		proto = "http"
+func (as *apiServer) getPublicURL(conf config.Prefix, pathPrefix string) string {
+	publicURL := conf.GetString(HTTPConfPublicURL)
+	if publicURL == "" {
+		proto := "https"
+		if !conf.GetBool(HTTPConfTLSEnabled) {
+			proto = "http"
+		}
```

Comment:
Should we consider the case where TLS is terminated at a load balancer or reverse proxy? In this case, I would think `HTTPConfTLSEnabled` would be `false` because FireFly isn't handling TLS itself, but the public URL would still need the `https` prefix. Maybe we just let `HTTPConfPublicURL` have the protocol built into it as well? In other words, maybe we should make `HTTPConfPublicURL` be the full URL?

Reply:
> maybe we should make HTTPConfPublicURL be the full URL

... This was the intention of the change... let me double check (I moved things around a couple of times and maybe I lost it)

Reply:
Yeah, think that's in place. If `publicURL` is set, we require it to be a full URL

Reply:
Nevermind. I see now that this is inside an `if` which only happens when `publcURL` is not set. This looks good.